here-naksha-lib-handlers/src/main/java/com/here/naksha/lib/handlers/DefaultStorageHandler.java

```diff
@@ -91,155 +107,258 @@ public DefaultStorageHandler(
+      RuntimeException re) {
+    if (previousAttempt == FIRST_ATTEMPT && re instanceof StorageNotInitialized) {
+      return retryDueToUninitializedStorage(ctx, storageImpl, collectionIds, wc);
+    }
+    logger.warn(
+        "No further reattempt strategy available for WriteCollections request (collectionIds: {}, previous attempt: {}. Rethrowing original exception",
+        join(", ", collectionIds),
+        previousAttempt);
+    throw re;
+  }
+
+  private @NotNull Result reattemptFeatureRequestForTheFirstTime(
+      final @NotNull NakshaContext ctx,
+      final @NotNull IStorage storageImpl,
+      final @NotNull List<String> collectionIds,
+      final @NotNull Request<?> request,
+      final @NotNull RuntimeException re) {
+    if (re instanceof StorageNotInitialized) {
+      return retryDueToUninitializedStorage(ctx, storageImpl, collectionIds, request);
+    } else if (indicatesMissingCollection(re)) {
+      try {
+        return retryDueToMissingCollection(ctx, storageImpl, collectionIds, request);
+      } catch (MissingCollectionsException mce) {
+        logger.warn("Retrying due to missing collection failed", mce);
+        return mce.toErrorResult();
+      }
+    } else {
+      throw re;
+    }
+  }
+
+  private @NotNull Result reattemptAfterStorageInitialization(
+      final @NotNull NakshaContext ctx,
+      final @NotNull IStorage storageImpl,
+      final @NotNull List<String> collectionIds,
+      final @NotNull Request<?> request,
+      final @NotNull RuntimeException re) {
+    if (indicatesMissingCollection(re)) {
+      try {
+        return retryDueToMissingCollection(ctx, storageImpl, collectionIds, request);
+      } catch (MissingCollectionsException mce) {
+        logger.warn("Retrying due to missing collection failed", mce);
+        return mce.toErrorResult();
       }
     } else {
       throw re;
     }
   }
 
+  private boolean indicatesMissingCollection(RuntimeException re) {
+    if (re.getCause() instanceof SQLException sqe) {
+      return MISSING_COLLECTION_SQL_ERROR_STATES.contains(sqe.getSQLState());
+    }
+    return false;
+  }
+
+  @NotNull
+  private Result retryDueToUninitializedStorage(
+      final @NotNull NakshaContext ctx,
+      final @NotNull IStorage storageImpl,
+      final @NotNull List<String> collectionIds,
+      final @NotNull Request<?> request) {
+    logger.info("Initializing Storage before reattempting write request.");
+    storageImpl.initStorage();
+    logger.info("Storage initialized");
+    return forwardRequestToStorage(ctx, request, storageImpl, collectionIds, ATTEMPT_AFTER_STORAGE_INITIALIZATION);
+  }
+
+  private Result retryDueToMissingCollection(
+      final @NotNull NakshaContext ctx,
+      final @NotNull IStorage storageImpl,
+      final @NotNull List<String> collectionIds,
+      final @NotNull Request<?> request) {
+    logger.warn("Collection not found for {}", collectionIds);
+    if (properties.getAutoCreateCollection()) {
+      logger.warn(
```

Comment:
this can be at info level

Reply:
done, moved couple of other log statements to `info` as well